client/my-sites/plans-features-main/components/test/plan-notice.tsx

```diff
@@ -111,32 +87,29 @@ describe( '<PlanNotice /> Tests', () => {
 	beforeEach( () => {
 		jest.resetAllMocks();
 
-		mGetDiscountByName.mockImplementation( () => false );
-		mUseMarketingMessage.mockImplementation( () => [ false, [], () => ( {} ) ] );
-		mIsCurrentPlanPaid.mockImplementation( () => true );
-		mIsCurrentUserCurrentPlanOwner.mockImplementation( () => true );
-		mIsRequestingSitePlans.mockImplementation( () => true );
-		mGetCurrentUserCurrencyCode.mockImplementation( () => 'USD' );
-		mUsePlanUpgradeCreditsApplicable.mockImplementation( () => 100 );
-		mUseDomainToPlanCreditsApplicable.mockImplementation( () => 100 );
-		mGetByPurchaseId.mockImplementation(
-			() =>
-				( {
-					isInAppPurchase: false,
-				} ) as Purchase
-		);
-		mIsProPlan.mockImplementation( () => false );
+		jest.mocked( getDiscountByName ).mockReturnValue( false );
+		jest.mocked( useMarketingMessage ).mockReturnValue( [ false, [], () => ( {} ) ] );
+		jest.mocked( isCurrentPlanPaid ).mockReturnValue( true );
+		jest.mocked( isCurrentUserCurrentPlanOwner ).mockReturnValue( true );
+		jest.mocked( isRequestingSitePlans ).mockReturnValue( true );
+		jest.mocked( getCurrentUserCurrencyCode ).mockReturnValue( 'USD' );
+		jest.mocked( usePlanUpgradeCreditsApplicable ).mockReturnValue( 100 );
+		jest.mocked( useDomainToPlanCreditsApplicable ).mockReturnValue( 100 );
+		jest.mocked( getByPurchaseId ).mockReturnValue( {
+			isInAppPurchase: false,
+		} as Purchase );
+		jest.mocked( isProPlan ).mockReturnValue( false );
 	} );
 
 	test( 'A contact site owner <PlanNotice /> should be shown no matter what other conditions are met, when the current site owner is not logged in, and the site plan is paid', () => {
-		mGetDiscountByName.mockImplementation( () => discount );
-		mUsePlanUpgradeCreditsApplicable.mockImplementation( () => 100 );
-		mIsCurrentPlanPaid.mockImplementation( () => true );
-		mIsCurrentUserCurrentPlanOwner.mockImplementation( () => false );
+		jest.mocked( getDiscountByName ).mockReturnValue( discount );
+		jest.mocked( usePlanUpgradeCreditsApplicable ).mockReturnValue( 100 );
+		jest.mocked( isCurrentPlanPaid ).mockReturnValue( true );
+		jest.mocked( isCurrentUserCurrentPlanOwner ).mockReturnValue( false );
 
 		renderWithProvider(
 			<PlanNotice
-				discountInformation={ { withDiscount: 'test', discountEndDate: new Date() } }
+				discountInformation={ { coupon: 'test', discountEndDate: new Date() } }
```

Comment:
Here I'm fixing a type error caught by TypeScript after removing a `ts-nocheck` comment. The `withDiscount` field was renamed to `coupon` in #96600 but forgot to update the tests.